packages/cfd/src/Containers/__tests__/cfd-password-modal.spec.js

```diff
@@ -0,0 +1,528 @@
+import React from 'react';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { WS, validPassword } from '@deriv/shared';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import CFDPasswordModal from '../cfd-password-modal';
+
+jest.mock('Stores/connect.js', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+jest.mock('@deriv/account', () => ({
+    SentEmailModal: jest.fn(({ should_show_sent_email_modal }) => (
+        <div>{should_show_sent_email_modal && <span>SentEmailModal</span>}</div>
+    )),
+}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Icon: jest.fn(({ icon }) => <div>{icon}</div>),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    getErrorMessages: jest.fn().mockReturnValue({
+        password_warnings: '',
+        password: jest.fn(),
+    }),
+    isDesktop: jest.fn().mockReturnValue(true),
+    WS: {
+        verifyEmail: jest.fn(),
+    },
+    validPassword: jest.fn().mockReturnValue(true),
+}));
+
+describe('<CFDPasswordModal/>', () => {
+    const mockFn = jest.fn();
+    const history = createBrowserHistory();
+    let modal_root_el;
+    const mock_props = {
+        account_title: '',
+        account_type: {},
+        account_status: {},
+        disableCFDPasswordModal: null,
+        email: '',
+        error_message: '',
+        error_type: '',
+        form_error: '',
+        getAccountStatus: null,
+        history: history,
+        is_eu: false,
+        is_fully_authenticated: false,
+        is_cfd_password_modal_enabled: false,
+        is_cfd_success_dialog_enabled: false,
+        is_dxtrade_allowed: false,
+        platform: '',
+        has_cfd_error: false,
+        landing_companies: {},
+        mt5_login_list: [],
+        cfd_new_account: {},
+        setCFDSuccessDialog: null,
+        setMt5Error: null,
+        submitMt5Password: null,
+        submitCFDPassword: null,
+    };
```

Comment:
I would suggest setting all callbacks in `mock_props` at once as `jest.fn()` or `jest.fn(() => <RETURN_VALUE>)` when some return value is expected (example: `jest.fn(() => true)` ), and also setting `mock_props` inside `beforeEach()` to avoid the need for defining the entire props object in each test case. So that in each test case, you only have to pass some of the props that actually affect the rendering result. Please check an example here: https://github.com/binary-com/deriv-app/pull/6181/files 🙂